lib/widgets/home.dart

```diff
@@ -0,0 +1,461 @@
+            constraints: const BoxConstraints(maxWidth: 600).tighten(height: 48),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceAround,
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: [
+                for (final (pageIndex, buildButton) in navigationButtonBuilders.indexed)
+                  Expanded(child: buildButton(pageIndex)),
+                Expanded(
+                  child: NavigationButton(
+                    icon: ZulipIcons.menu, tooltip: 'Menu', selected: false,
+                    onPressed: () => showMenu(context))),
+              ])))));
+  }
+}
+
+const Duration _bottomSheetEnterDuration = Duration(milliseconds: 250);
+const Duration _bottomSheetExitDuration = Duration(milliseconds: 200);
+
+class _PausableAnimationController extends AnimationController {
+  _PausableAnimationController({
+    required super.vsync,
+    super.duration,
+    super.reverseDuration,
+  });
+
+  bool paused = false;
+
+  @override
+  set value(double newValue) {
+    if (paused) {
+      return;
+    }
+    super.value = (newValue);
+  }
+}
+
+void showMenu(BuildContext context) async {
+  final store = PerAccountStoreWidget.of(context);
+  final designVariables = DesignVariables.of(context);
+  final menuItems = <Widget>[
+    // Search
+    // const SizedBox(height: 8),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: true, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    _MenuButton(selected: false, pageContext: context),
+    // Inbox
+    // Recent conversations
+    // Mentions
+    // Starred messages
+    // Drafts
+    // Direct messages
+    // Streams
+    // Users
+    // My profile
+    // Set my status
+    // const SizedBox(height: 8),
+    // Settings
+    // Notifications
+    const SizedBox(height: 8),
+    const _VersionInfo(),
+  ];
+  final animationController = _PausableAnimationController(
+    vsync: Navigator.of(context),
+    duration: _bottomSheetEnterDuration,
+    reverseDuration: _bottomSheetExitDuration,
+  );
+  bool handleScroll(ScrollNotification notification) {
+    animationController.paused = notification.metrics.extentBefore != 0;
+    return false;
+  }
```

Comment:
wip; nav: Disable bottom sheet dragging when scrolled

I think the behavior before this commit is fine and actually a bit nicer. So let's leave out this complication.

I gather you're referring to this line in the Figma:
> It would be nice if swipe down can close the menu, but it is important to do that only, when content is scrolled to its top position. And it isn’t just a swipe gesture but full menu dragging 

But I think the thing Vlad means is just: if you've scrolled down in the content and start swiping down, it should scroll you back up in the content (just like normal for a scrollable) instead of having the menu start closing.

When I run with the version before this commit, that all works fine: if you're scrolled down, then swiping anywhere within the scrollable acts on the scrollable before doing anything to the menu as a whole. If you touch the header and drag that, then that does control the menu… but I think that's fine and actually good. After this commit, attempting to drag the header doesn't do anything if the options are scrolled, which to me feels oddly inert like something's not working.

Reply:
Yeah, I agree to leaving this out.

> It would be nice if swipe down can close the menu

I think to fully implement this (out of scope) feature, we will need to allow over scrolling the inner content (swiping down the menu items and reaching the top of the menu) to drag the bottom sheet; once you scroll pass a certain extent the menu collapses, as it does when you control it by dragging the header. I guess the menu would otherwise rebound back in position if you release it.

It makes sense to leave this out because it will be quite complicated. An implementation might have to listen to notifications like `OverscrollNotification`, `ScrollEndNotification` and replicate the drag handlers from `_BottomSheetState` and manipulate the animation controller.